[Hua Kun] iP

tanhuakun opened the pull request:
Better Duke
> Best Duke! - Me ([source](https://github.com/nus-cs2103-AY2122S1/ip/pull/194#issue-717302301))

- Text-based!
- Great!

All you need to do is,

1. Download it from [here!](https://github.com/tanhuakun/ip/releases/tag/v0.1)
2. Double-click the Jar file
3. Enjoy.

It is **FREE!**

Features:

- [X] Managing tasks
- [ ] Sharing tasks (coming soon)

Here is Duke's main method:
```
public static void main(String[] args) {
    Duke duke = new Duke();
    duke.start();
}
```

Commit:
Add Gradle support

tanhuakun committed:
Level 1, added Greet Echo Exit

tanhuakun committed:
Added Level-2 Add, List

tanhuakun committed:
Added Level-3 mark as done

tanhuakun committed:
Added Level 4 Todos, events and deadlines

tanhuakun committed:
Modified testing files

tanhuakun committed:
Added exception handling

tanhuakun committed:
Added delete function for level-6

tanhuakun committed:
Added saving for Level-7

tanhuakun committed:
Added date-time for level-8.

tanhuakun committed:
Merge branch 'branch-Level-8'

tanhuakun committed:
Fixed compile error undetected during merge

tanhuakun committed:
Added A-MoreOOP

tanhuakun committed:
Created Packages

tanhuakun committed:
Added tests

tanhuakun committed:
Add A-JavaDoc requirement

tanhuakun committed:
Format to coding standard

tanhuakun committed:
Add find feature

tanhuakun committed:
Merge branch 'branch-A-CodingStandard'

tanhuakun committed:
Merge branch 'branch-Level-9'

tanhuakun committed:
Fix compile error

guangxuan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,48 @@
+package duke;
+
+import duke.tasks.*;
+import duke.command.*;
+import duke.exceptions.*;
```
Remember to expand your imports
https://se-education.org/guides/conventions/java/intermediate.html#package-and-import-statements

guangxuan commented on src/main/java/duke/FileController.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    /**
+     * Returns the whole contents of the file in a single String
+     *
+     * @return contents of file specified in this object
```
Can add punctuations behind descriptions, parameters, and returns.

guangxuan commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,81 @@
+    ArrayList<Task> storedList;
+
+    /**
+     * Constructor for tasklist
+     * @param string a string representation of tasks
```
``` suggestion
     * Constructor for tasklist
     * 
     * @param string a string representation of tasks
```

guangxuan commented on src/main/java/duke/tasks/TaskList.java:
```diff
@@ -0,0 +1,81 @@
+        });
+    }
+
+    public void delete(int index) {
+
+    }
```
Seems to be a repeated function with remove.

guangxuan reviewed (commented):
LGTM. A couple of javadocs conventions to be followed.
Good classification of exceptions, also nice abstraction of commands.
Parsing of file was slightly too complicated.

tanhuakun committed:
Fix things from previous week

tanhuakun committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tanhuakun committed:
Add Gradle and Checkstyle

MarcusTXK commented on src/main/java/duke/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,23 @@
+package duke.command;
+
+import duke.*;
```
Perhaps expand this out instead of using a wildcard import?

One useful setting I use in IntelliJ is to go to `Preferences > Editor > Code Style > Java > Imports tab` and set `Class count to use import with '*'` and `Names count to use static import with '*'` to a very high value (I personally set it at 100).

MarcusTXK commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,38 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+
```
Consider adding a descriptive header for the Event class? I noticed that you may have missed adding header comments for some other classes as well, so perhaps it would be good to consitently write descriptive header comments for all public classes/methods.

MarcusTXK commented on src/main/java/duke/tasks/TaskUtils.java:
```diff
@@ -0,0 +1,67 @@
+        }
+        boolean isDone = segments[1].equals(DONE_MARKER);
+        Task toReturn;
+        switch (segments[0]) {
+            case "T":
+                toReturn = new Todo(segments[2]);
+                break;
```
Perhaps remove the indentation in the switch statement, as there should be no indentation for `case` clauses. 

One useful way is to configure Intellij's settings so they can handle this indentation for you. Go to `Settings > Editor > Code Style > Java`, select tab `Wrapping and Braces`, scroll down the list to `switch` statement, uncheck the option `Indent 'case' branches`.

```suggestion
        switch (segments[0]) {
        case "T":
            toReturn = new Todo(segments[2]);
            break;
```

MarcusTXK reviewed (commented):
Overall, I found your code easy to read as the names followed the coding standard, and the code structure was good, nice job!

LGTM, almost. Just a few minor parts to fix.

tanhuakun committed:
Add support for GUI level-10. Removed UI and exit command

tanhuakun committed:
Remove var declaration in dialogbox

tanhuakun committed:
Fix some checkstyle changes

tanhuakun committed:
Fulfill requirements for A-CodeQuality

Requirements state that code must adhere to code quality guidelines.

Many cases of magic Strings were refactored and comments were added to
improve readability.

These changes were done to ensure that code is of good quality.

Code quality guide can be found on CS2103 website.

tanhuakun committed:
Fulfill requirements for A-Assertions

Assertions enable us to check various assumptions. Having
assertions are good.

Assert statements were added to some files.

Having assert statements help with asserting assumptions.

More info about assertions can be found on the CS2103 website.

tanhuakun committed:
Merge pull request #1 from tanhuakun/branch-A-Assertions

Fulfill requirements for A-Assertions

tanhuakun committed:
Merge branch 'master' into branch-A-CodeQuality

tanhuakun committed:
Merge pull request #2 from tanhuakun/branch-A-CodeQuality

Fulfill code quality requirements

tanhuakun committed:
Add workflow

tanhuakun committed:
Move workflow folder

tanhuakun committed:
Removed workflow

tanhuakun committed:
Create javaCI.yml

tanhuakun committed:
Add comment to test github CI

tanhuakun committed:
Merge branch 'master' of https://github.com/tanhuakun/ip

tanhuakun committed:
Update build.gradle to not use local path

tanhuakun committed:
Remove checkstyle from build.gradle

tanhuakun committed:
Remove checkstyle for real

tanhuakun committed:
Change build.gradle checkstyle function to not run

tanhuakun committed:
Add support for new recurring task class

tanhuakun committed:
Merge branch 'branch-B-RecurringTasks'

tanhuakun committed:
Improve GUI and add more comments

tanhuakun committed:
Add first version of user guide

tanhuakun committed:
Update user guide

tanhuakun committed:
Update user guide

tanhuakun committed:
Add Ui.png screenshot
